### ISSUE TITLE
An item with the same key has already been added.

### ISSUE THREAD

**Issue by myaseer** [nopSolutions/nopCommerce#2860]
nopCommerce version: 4.00

Expected behavior:
In admin panel of nopCommerce need to check for existing category name before adding/importing categories.

Actual behavior:
Its throws an item with the same key has already been added error, while importing products from excel file. When I look at the code it was this particular line causing that issue.

var allCategories = _categoryService.GetAllCategories(showHidden: true).ToDictionary(c => c.GetFormattedBreadCrumb(_categoryService), c => c);

That's because it allows categories to be added with the same category name, and while importing a products from excel loading all categories into a dictionary with category name as key.

Steps to reproduce the problem: 
1. Login to admin panel
2. Add two categories with same name.
3. Import Products from an excel file.

**Comment by AndreiMaz** [comment]
@myaseer Why do you need two categories with the same name? I mean at the same level?

**Comment by myaseer** [comment]
Hi @AndreiMaz, I don't need categories with the same name, which doesn't make any sense. 
What actually happen, while importing categories from excel there was a duplicate entry in that file, It didn't throw any warning or an error message so I left assuming that everything goes well. 

However while importing a bunch of products from an excel file, I got an error message saying that "An item with the same key has already been added." By looking at the error message I assumed there must be an duplicate entry in the products. Started searching for possible duplicate entry in that excel file and I couldn't find any. I was wondering what could possibly go wrong. I couldn't figure it out what's going wrong until I had look through the code to find an error.
